### DISCUSSION TITLE
LAB D SDRAM 問題

### DISCUSSION THREAD

**Discussion by PatriChou** [bol-edu/HLS-SOC-Discussions#198] [Q&A]
老師好 各位同學們好,

目前我們這組想要去嘗試做Lab D SDRAM的優化，我們發現如果Controller 在Read State 時連續對SDRAM傳送多筆的Request，SDRAM就會連續吐出多筆Data ，如下圖所示

![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145217252/db19521b-2c5c-48ed-91e5-7fd2ffbf0381)

想請問這麼做是可以的嗎? 還是我們需要按照IDLE -> READ -> READ_RES -> IDLE 去讀出每一筆Data。

謝謝各位

**Comment by Yanlin900414** [comment]
可以在Read階段一直Request它，但我的版本不知道為何如果不包含進3個T的等待時間的話，後續會出問題。因此我最終設計是Read->Wait 3T ->Read ....直到我要Prefetch的那個FIFO收滿八個指令

**Reply by PatriChou** [reply]
原來如此 所以在Read階段一直Request是可以的 非常感謝您的回答

**Comment by boledulab** [comment]
Refer to the datasheet (https://github.com/bol-edu/caravel-soc_fpga-lab/blob/main/lab-sdram/sdram/256Mb_sdr.pdf). Page#52. SDRAM device allows pipeline command access. 

![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/98332019/4b4457b0-fa43-4a7a-a227-7e7e3705c16b)

**Reply by PatriChou** [reply]
謝謝老師 那這樣我明白了
